griptape/tools/shopify_client/tool.py

```diff
@@ -0,0 +1,66 @@
+from textwrap import dedent
+from typing import Optional
+from urllib.parse import urljoin
+import schema
+from schema import Schema, Literal
+from attr import define, field
+from griptape.tools import BaseTool
+from griptape.utils.decorators import activity
+from griptape.artifacts import BaseArtifact, TextArtifact, ErrorArtifact
+
+
+@define
+class ShopifyClient(BaseTool):
+    storename: str = field(kw_only=True)
+    storefront_access_token: str = field(kw_only=True)
+    shopify_endpoint: str = field(default="myshopify.com/api/2023-07/graphql.json")
```

Comment:
It says "not found" when I hit the endpoint in the browser.

Reply:
Missing `kw_only=True`.

Reply:
Should we call it something like `schema_endpoint` instead?

Reply:
@vasinov it's just a component of the full url. From the [Shopify curl docs](https://shopify.dev/docs/api/storefront#authentication):
`https://{store}.myshopify.com/api/2021-07/graphql.json`